sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -159,6 +159,7 @@ class Analyzer(
     Batch("Resolution", fixedPoint,
       ResolveTableValuedFunctions ::
       ResolveRelations ::
+      ResolveOutputRelation ::
```

Comment:
I had to move this higher in the batch so that outputs were resolved before the overwrite expression (see analysis tests for `OverwriteByExpression`). Otherwise, the expression would be resolved against the overwrite's child query.

Another approach is to change how the expression's attributes are resolved to use the table instead of the query. But changing expression resolution to use something other than children seemed too intrusive.

Reply:
Thanks for the information about how to solve this. I'll update it.